core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/Action.scala

```diff
@@ -208,195 +199,243 @@ abstract class SingleResourceAction[T <: ObjectResource] extends ResourceAction[
-        .map(existingResource =>
-          editor.updateMetadata(resource, resource.metadata.copy(resourceVersion = existingResource.metadata.resourceVersion))
-        )
-      res <- resourceVersionUpdated
-        .map { resourceToUpdate =>
-          if (predicateForUpdate(existing, resourceToUpdate)) {
-            recoverFromError(client.updateStatus(resourceToUpdate), client, retries - 1, executeUpdateStatus)
-          } else {
-            Action.log.info(s"Ignoring status update for resource ${resource.metadata.name}")
-            Future.successful(resource)
-          }
-        }
-        .getOrElse(Future.successful(resource))
-    } yield res
-}
+    } yield new UpdateStatusAction(result)
 
-object DeleteAction {
-  def apply[T <: ObjectResource](resource: T, resourceDefinition: ResourceDefinition[T]) =
-    new DeleteAction(resource.metadata.name, resource.metadata.namespace, resourceDefinition)
+  def executeUpdateStatus(client: KubernetesClient, retries: Int = 60)(implicit sys: ActorSystem, ec: ExecutionContext): Future[T] = {
+    val existing = Try {
+      client
+        .resource(resource)
+        .inNamespace(namespace.getOrElse(client.getNamespace))
+    }.toOption
+
+    //    TODO: port the update ..
+    //    for {
+    //      resourceVersionUpdated = existing
+    //        .map(existingResource =>
+    //          editor.updateMetadata(resource, resource.metadata.copy(resourceVersion = existingResource.metadata.resourceVersion))
+    //        )
+    //      res <- resourceVersionUpdated
+    //        .map { resourceToUpdate =>
+    //          if (predicateForUpdate(existing, resourceToUpdate)) {
+    //            recoverFromError(client.resource(resource).updateStatus(resourceToUpdate), client, fabric8Client, retries - 1, executeUpdateStatus)
+    //          } else {
+    //            Action.log.info(s"Ignoring status update for resource ${resource.metadata.name}")
+    //            Future.successful(resource)
+    //          }
+    //        }
+    //        .getOrElse(Future.successful(resource))
+    //    } yield res
+    ???
+  }
 }
 
 /**
  * Captures deletion of the resource.
  */
-final case class DeleteAction[T <: ObjectResource](
-    val resourceName: String,
-    _namespace: String,
-    implicit val resourceDefinition: ResourceDefinition[T]
+final case class DeleteAction[T <: HasMetadata](val resourceName: String, _namespace: String)(
+    implicit ct: ClassTag[T]
 ) extends ResourceAction[T] {
+  import io.fabric8.kubernetes.api.model.{ DeletionPropagation, ObjectMetaBuilder }
+  import io.fabric8.kubernetes.client.utils.Serialization
+
   val name      = "delete"
   val namespace = Some(_namespace)
   override def executing =
     s"Deleting $resourceName resource"
   override def executed =
     s"Deleted $resourceName resource"
 
+  private val log = LoggerFactory.getLogger(this.getClass.getName)
+
+  private val meta = {
+    val default = {
+      Serialization
```

Comment:
@RayRoestenburg fyi:
This looks like a somehow "better way" than using default constructors (this way all the properties injected via Json got populated)